blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/AsyncWriter.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2016 Rackspace.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.io.datastax;
+
+import com.datastax.driver.core.ResultSetFuture;
+import com.rackspacecloud.blueflood.rollup.Granularity;
+import com.rackspacecloud.blueflood.types.Locator;
+import com.rackspacecloud.blueflood.types.Rollup;
+
+/**
+ * An interface describing the behavior of classes who can
+ * asynchronously insert a metric to cassandra. This only
+ * applies to classes that use datastax driver.
+ */
+public interface AsyncWriter {
```

Comment:
I need it in Datastax's PreaggregatedMetrics. I am keeping a map of rollupType => corresponding Async Writer. 
https://github.com/rackerlabs/blueflood/pull/663/files/f4170585379e101e680979d9051da390b9aeac1a#diff-579f34fdebe138b9a4ab835a28731440R47

When PreaggregatedMetrics gets a list of IMetrics objects, it needs to be able to call the right AsyncWriter to write that to the CFs. It is a Datastax specific interface only, thus it lives in the datastax package. I am not writing re-writing the Astyanax logic to do the same, because eventually we will be getting rid of them.